Arrow Validation Implementation

char2cs opened the pull request:
### Changes

**New Features:**
- Added arrow validation with JSON schema support
- Implemented version compatibility checking for arrow manifests
- Created versioned schema registry (v1 for Arrow and Quiver)
- Added validator component with detailed error reporting

**Infrastructure:**
- Implemented parser for manifest metadata extraction
- Added reader component for Arrow and Quiver manifests
- Created schema mappers for v1 Arrow and Quiver specifications

**Dependencies:**
- Added `gojsonschema` for JSON schema validation

### Testing
- Added comprehensive unit tests for all new components
- Validated version compatibility logic
- Maintained test coverage requirements

### Technical Details
- Follows Clean Architecture principles with proper layer separation
- Schema registry supports multiple manifest versions
- Validator provides detailed error reporting for validation failures

CasajoanaGines committed:
Implement ValidateArrow and unit test for version compatibility

char2cs committed:
chore(dependencies): add gojsonschema and related indirect dependencies

This commit updates the go.mod file to include the gojsonschema package and its indirect dependencies (gojsonpointer and gojsonreference). Additionally, it removes unused translator components and their associated tests, streamlining the infrastructure layer.

char2cs committed:
Merge branch 'develop' into feature/validate-arrow

char2cs committed:
refactor(translator): clean up whitespace and formatting in translator files

This commit removes unnecessary blank lines and standardizes the formatting of context parameters across the Translator interface and its implementations. Additionally, it adjusts the ValidationResult and ValidationError structs for consistent field alignment.

char2cs committed:
chore(dependencies): remove duplicate testify dependency in go.mod

This commit cleans up the go.mod file by removing the duplicate entry for the testify package, ensuring a more organized and maintainable dependency list.

Lucacux reviewed (dismissed):
**Pros**

Adding JSON Schema–based validation for Arrow and Quiver is a solid improvement. It makes the data contracts explicit and easier to evolve.

The new structure under internal/infrastructure/translator/schemas (arrow/quiver v1, registry, types) is clean and easy to follow.

Using gojsonschema keeps the validation logic focused and avoids maintaining custom schema-handling code.

Cleaning up the old translator code and routing everything through the new validator/registry makes the translation flow more consistent.

**Suggestions**

In validator_test.go, it would be helpful to add stronger negative tests: invalid payloads, missing fields, incorrect or unsupported versions, etc. Most tests currently cover the happy path, which could hide issues if the schemas change later.

registry.go is now the central piece that maps types and versions to their schemas. It may be worth tightening the tests around unknown types, unknown versions, and mismatched mappings so that inconsistencies fail fast instead of going unnoticed

char2cs committed:
refactor(infrastructure): update models and remove unused translator components

This commit modifies the Infrastructure layer by changing the Translator field in the Infrastructure struct to a pointer and updating the instantiation of the Translator. It also refactors the Netbridge interface to use the new netbridge models, ensuring consistency across the codebase. Additionally, it removes the Translator implementation and its associated tests, streamlining the infrastructure layer.

char2cs committed:
Merge branch 'develop' into feature/validate-arrow

Resolved conflicts:
- go.mod/go.sum: Merged dependencies from both branches
- netbridge: Updated to use ForwardingStatusDisabled for ReversePort
- requirements: Added gopsutil implementation for CPU/Memory/Disk validation
- Maintained arrow.Requirement and shared.OS models from this branch

char2cs committed:
test(models): add validation tests for Arrow, Requirement, Variable, and PortRule

This commit introduces comprehensive validation tests for the Arrow, Requirement, Variable, and PortRule models. Each model's validation logic is thoroughly tested with various scenarios, ensuring that errors are correctly raised for invalid inputs and that valid configurations are accepted. This enhancement improves the robustness of the validation framework and increases overall test coverage.

char2cs committed:
Merge branch 'develop' into feature/validate-arrow

char2cs committed:
test(watcher): add logging tests for Debug, Info, Warn, and Error functions

This commit introduces unit tests for the logging functions in the Watcher service. The tests cover Debug, Info, Warn, Error, and formatted Error logging, ensuring that each function behaves as expected. Additionally, tests for unforeseen logging calls are skipped to prevent process termination during testing. This enhancement improves test coverage and reliability of the logging functionality.

char2cs committed:
fix(infrastructure): enhance file reading to prevent directory traversal

This commit updates the readFile function in the translator package to include path validation, ensuring that directory traversal attacks are prevented. The function now cleans the file path and checks for invalid patterns before attempting to read the file. This enhancement improves the security of file handling within the infrastructure layer.

char2cs committed:
style(infrastructure): clean up whitespace in readFile function

This commit removes unnecessary whitespace in the readFile function within the translator package. The changes enhance code readability without altering functionality, maintaining the existing path validation logic to prevent directory traversal attacks.

AgustinGil21 commented on internal/models/arrow/method.go:
```diff
@@ -0,0 +1,78 @@
+		return fmt.Errorf("action type cannot be empty")
+	}
+
+	switch a.Type {
```
No entendí este switch

AgustinGil21 replied:
Entiendo que evalua pero no entiendo porque se usa el switch ahí.

char2cs replied:
Ahh, esta evaluando si es alguna de la const o no. Si no falla. 

Me parecio mas facil que hacer un IF con muchos OR ( || )

AgustinGil21 replied:
Ahh listo, perfecto entonces

AgustinGil21 reviewed (commented):
Ahí te deje un solo comentario para que lo revises.
